nav2_motion_primitives/include/nav2_motion_primitives/motion_primitive.hpp

```diff
@@ -91,6 +104,19 @@ class MotionPrimitive
   }
 
 protected:
+  // The node to use for publishers, subscribers and logging
+  rclcpp::Node::SharedPtr node_;
+
+  // Proxy for getting the robot pose
+  std::shared_ptr<nav2_robot::Robot> robot_;
+
+  // Proxy to the World Model
+  nav2_world_model::WorldModelClient world_model_;
```

Comment:
This class never uses the world_model_. Seems likely the wrong place for it.

Reply:
This is the base class for the motion primitives. Trying to avoid duplicating this line across all primitives.